lib/exports.ts

```diff
@@ -3,19 +3,77 @@ import mdast from './mdast';
 import { isMDXEsm } from '../processor/utils';
 import { MdxjsEsm } from 'mdast-util-mdx';
 
-const EXPORT_NAME_REGEX = /export\s+(?:const|let|var|function)\s+(\w+)/;
+
 
 const exports = (doc: string) => {
   const set = new Set<string>();
 
   visit(mdast(doc), isMDXEsm, (node: MdxjsEsm) => {
-    if (node.value?.match(EXPORT_NAME_REGEX)) {
-      const [, name] = node.value.match(EXPORT_NAME_REGEX);
-      set.add(name);
+    // Once inside an mdxjsEsm node, we need to check for one or more declared exports within data.estree.body
+    // This is because single newlines \n are not considered as a new block, so there may be more than one export statement in a single mdxjsEsm node
+    const body = node.data?.estree.body;
+    if (!body) return;
+
+    for (const child of body) {
+        if (child.type === 'ExportNamedDeclaration') {
+        // There are three types of ExportNamedDeclaration that we need to consider: VariableDeclaration, FunctionDeclaration, ClassDeclaration
+        const declaration = child.declaration;
+        // FunctionDeclaration and ClassDeclaration have the same structure 
+        if (declaration.type !== 'VariableDeclaration') {
+          // Note: declaration.id.type is always 'Identifier' for FunctionDeclarations and ClassDeclarations
+          set.add(declaration.id.name);
+        }
+        else  {
+          const declarations = declaration.declarations;
```

Comment:
I'm not sure how closely this AST lines up with the AST that ESLint uses but for `export { dogs, cats }` they would be under a `specifiers` array.

![Screenshot 2025-01-29 at 9 25 39 AM](https://github.com/user-attachments/assets/95872853-c4e3-4289-acde-38f49013cd65)

Reply:
https://github.com/eslint/eslint/blob/main/package.json#L112

Pretty sure they are using the same AST. 👍🏼

Reply:
I don't think mdx supports that syntax?!?!

![Screenshot from 2025-01-29 09-56-17](https://github.com/user-attachments/assets/8e3b18cb-5636-4c84-9025-542332f36777)

Reply:
oh weird